skimage/segmentation/_remap.pyx

```diff
@@ -0,0 +1,26 @@
+# distutils: language = c++
+
+from libcpp.unordered_map cimport unordered_map
+from cython.parallel import prange
+import numpy as np
+cimport cython
+from .._shared.fused_numerics cimport np_numeric, np_anyint
+
+@cython.boundscheck(False)  # Deactivate bounds checking
+@cython.wraparound(False)   # Deactivate negative indexing
+def _map_array(np_anyint[:] inarr, np_numeric[:] outarr,
+               np_anyint[:] inval, np_numeric[:] outval):
+    # build the map from the input and output vectors
+    cdef size_t i, n_map, n_array
+    cdef unordered_map[np_anyint, np_numeric] lut
+    n_map = inval.shape[0]
+    for i in range(n_map):
+        lut[inval[i]] = outval[i]
```

Comment:
Should we check that `inval` only contains positive values?

Reply:
Actually, `lut` is an unordered_map (ie a dictionary), so it doesn't matter. This will work with negative values just as well as with positive ones.